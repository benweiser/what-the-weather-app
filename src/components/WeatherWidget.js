import { PropTypes } from "prop-types";
import React from "react";
import Button from "./Button";
import Input from "./Input";

class WeatherWidget extends React.PureComponent {
  state = {
    zipCode: ""
  };

  handleChange = e => {
    this.setState({ zipCode: e.target.value });
    e.preventDefault();
  };

  handleSubmit = e => {
    this.props.fetchWeather(this.state.zipCode);
    e.preventDefault();
    console.log("this state", this.state);
  };

  render() {
    console.log("this props", this.props);
    console.log("this state", this.state);
    return (
      <form onSubmit={this.handleSubmit}>
        <Input
          placeholder="Enter city name"
          onChange={this.handleChange}
          value={this.state.value}
          aria-label="Enter city name"
        />
        <Button type="submit">Submit Data</Button>
      </form>
    );
  }
}

WeatherWidget.propTypes = {
  fetchWeather: PropTypes.func.isRequired
};

export default WeatherWidget;
