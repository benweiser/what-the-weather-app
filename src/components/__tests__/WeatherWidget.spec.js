import renderer from 'react-test-renderer';
import React from 'react';
import WeatherWidget from '../WeatherWidget';

describe('Weather Widget Snapshot', () => {
  test('renders', () => {
    const component = renderer.create(<WeatherWidget />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
