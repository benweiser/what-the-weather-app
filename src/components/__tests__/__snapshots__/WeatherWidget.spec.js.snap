// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Widget Snapshot renders 1`] = `
<div>
  <input
    aria-label="Enter city name"
    className="Input"
    placeholder="Enter city name"
  />
  <button
    className="Button"
    onClick={[Function]}
  >
    Submit Data
  </button>
</div>
`;
